File: 160715/math/ruby/ten.rb
# 10. 请设计一个算法，计算n的阶乘有多少个尾随零。给定一个n，请返回n的阶乘的尾零个数。保证n为正整数。
num = 1000

def number_class(num)
  init_num = 1
  zero_count = 0
  (2..num).each do |m|
    init_num *= m
  end
  class_num_str = init_num.to_s.reverse
  (0...class_num_str.length).each do |c|
    new_num = class_num_str[c].to_i
    zero_count += 1 if new_num == 0
    break if new_num != 0
  end
  zero_count
end

result = number_class(num)

puts "#{num} 的阶层的尾零个数为 #{result} 个"

File: 160715/math/java/ten.java
// 10. 请设计一个算法，计算n的阶乘有多少个尾随零。给定一个n，请返回n的阶乘的尾零个数。保证n为正整数。

public class ten{
  public static void main(String[] args) {
    long num = 20;
    long result = numberClass(num); 
    System.out.println(num + "的阶层的尾零个数为：" + result);
  }

  public static long numberClass(long num){
    long zeroCount = 0; 
    long initNum = 1;
    String judgeStr = "";
    for (long i = 2; i <= num; i++) {
      initNum *= i; 
    }
    String numStr = String.valueOf(initNum);
    for (int j = numStr.length() - 1; j >= 0; j--) {
      judgeStr += numStr.charAt(j);
    }
    for (int k = 0; k < judgeStr.length(); k++) {
      String str = String.valueOf(judgeStr.charAt(k));
      if ("0".equals(str)) {
        zeroCount += 1; 
      }
      if (!"0".equals(str)) 
        break;
    }
    return zeroCount;
  }
}